src/wp-includes/class-wp-scripts.php

```diff
@@ -471,21 +505,21 @@ public function add_inline_script( $handle, $data, $position = 'after' ) {
 	 *                         Must be lowercase.
 	 * @param string $position Optional. Whether to add the inline script
 	 *                         before the handle or after. Default 'after'.
-	 * @param bool   $display  Optional. Whether to print the script
-	 *                         instead of just returning it. Default true.
-	 * @return string|false Script on success, false otherwise.
+	 * @param bool   $display  Optional. Whether to print the script tag
+	 *                         instead of just returning the script data. Default true.
+	 * @return string|false Script data on success, false otherwise.
 	 */
 	public function print_inline_script( $handle, $position = 'after', $display = true ) {
 		_deprecated_function( __METHOD__, '6.3.0', 'WP_Scripts::get_inline_script_data() or WP_Scripts::get_inline_script_tag()' );
-		if ( $display ) {
-			$output = $this->get_inline_script_tag( $handle, $position );
-			echo $output;
-		} else {
-			$output = $this->get_inline_script_data( $handle, $position );
-		}
+
+		$output = $this->get_inline_script_data( $handle, $position );
 		if ( empty( $output ) ) {
 			return false;
 		}
+
+		if ( $display ) {
+			echo $this->get_inline_script_tag( $handle, $position );
+		}
 		return $output;
 	}
```

Comment:
This change restores the legacy behavior for this deprecated method, where it's supposed to return the script data always, not the script tag.